### PR TITLE
Add spans to provide more information profile gaps

### PR DESCRIPTION
**Pull request by cwfitzgerald**
**Description**

Trying to fill in some information gaps in profiles.
- If there is a long running api call, mark it specifically to indicate what is the actual root performance loss.
- Add spans in places where there there was a decent amount of work happening, but there was no explanation as to why.

Did this for dx12 and vulkan as these are the platforms I have access to.

**Testing**

Inspected logs in tracy

### CODE REVIEW

File: wgpu-core/src/command/render.rs [hunk 1]
```diff
@@ -729,6 +729,7 @@ impl<'a, A: HalApi> RenderPassInfo<'a, A> {
     ) -> Result<StatefulTrackerSubset, RenderPassErrorInner> {
         profiling::scope!("finish", "RenderPassInfo");
         unsafe {
+            profiling::scope!("end_render_pass");
```
Comment (kvark) [inline wgpu-core/src/command/render.rs:732]: there is already a scope for this, why do we need the nested one?

File: wgpu-core/src/device/life.rs [hunk 2]
```diff
@@ -128,35 +128,65 @@ impl<A: hal::Api> NonReferencedResources<A> {
     }
 
     unsafe fn clean(&mut self, device: &A::Device) {
-        for raw in self.buffers.drain(..) {
-            device.destroy_buffer(raw);
+        {
+            profiling::scope!("destroy_buffers");
```
Comment (kvark) [inline wgpu-core/src/device/life.rs:132]: let's guard this with `if !self.buffers.is_empty()` and everything below the same way

File: wgpu-hal/src/dx12/adapter.rs [hunk 3]
```diff
@@ -44,21 +44,27 @@ impl super::Adapter {
         library: &Arc<native::D3D12Lib>,
         instance_flags: crate::InstanceFlags,
     ) -> Option<crate::ExposedAdapter<super::Api>> {
+        profiling::scope!("expose adapter");
```
Comment (kvark) [inline wgpu-hal/src/dx12/adapter.rs:47]: this thing should live in `wgpu-core` if it doesn't already? No point in adding this scope to each backend

File: wgpu-hal/src/dx12/command.rs [hunk 4]
```diff
@@ -767,9 +767,11 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
                 };
                 self.temp.barriers.push(barrier);
             }
+            profiling::scope!("ID3D12GraphicsCommandList::ResourceBarrier");
```
Comment (kvark) [inline wgpu-hal/src/dx12/command.rs:770]: This scope would cover more things than just the next line. If you want this scope, let's do `if !self.temp.barriers.is_empty() { ... }`

**Review by kvark — COMMENTED** [review]
This looks useful, thank you!
Just a few things that we may fix

File: wgpu-core/src/command/render.rs (hunk 1 above)
Comment (cwfitzgerald) [reply wgpu-core/src/command/render.rs:732]: We don't, must have left that in while investigating.

File: wgpu-hal/src/dx12/adapter.rs (hunk 3 above)
Comment (cwfitzgerald) [reply wgpu-hal/src/dx12/adapter.rs:47]: Another investigation artifact.

**Review by kvark — APPROVED** [review]

**Pull request merged**